ci/makefiles/android.mk

```diff
@@ -0,0 +1,78 @@
+# Downloads and installs the Android SDK depending on supplied platform: darwin or linux
+
+# We must provide a platform (darwin or linux) and we need JAVA_HOME defined
+ifndef target_os
+    $(error target_os is not set...aborted!)
+endif
+
+# Those android NDK/SDK variables can be override when running the file
+ifndef ANDROID_NDK_VERSION
+    ANDROID_NDK_VERSION=17c
+endif
+ifndef ANDROID_SDK_TOOLS_VERSION
+    ANDROID_SDK_TOOLS_VERSION=4333796
+endif
+ifndef ANDROID_SDK_BUILD_TOOLS_VERSION
+	ANDROID_SDK_BUILD_TOOLS_VERSION=28.0.2
+endif
+ifndef ANDROID_HOME
+    ANDROID_HOME=$(HOME)/.android
+endif
+ifndef ANDROID_API_LEVEL
+    ANDROID_API_LEVEL=27
+endif
+
+ANDROID_SDK_HOME=$(ANDROID_HOME)/android-sdk
+ANDROID_SDK_TOOLS_ARCHIVE=sdk-tools-$(target_os)-$(ANDROID_SDK_TOOLS_VERSION).zip
+ANDROID_SDK_TOOLS_DL_URL=https://dl.google.com/android/repository/$(ANDROID_SDK_TOOLS_ARCHIVE)
+
+ANDROID_NDK_HOME=$(ANDROID_HOME)/android-ndk
+ANDROID_NDK_FOLDER=$(ANDROID_HOME)/android-ndk-r$(ANDROID_NDK_VERSION)
+ANDROID_NDK_ARCHIVE=android-ndk-r$(ANDROID_NDK_VERSION)-$(target_os)-x86_64.zip
+ANDROID_NDK_DL_URL=https://dl.google.com/android/repository/$(ANDROID_NDK_ARCHIVE)
+
+$(info Target install OS is          : $(target_os))
+$(info Android SDK home is           : $(ANDROID_SDK_HOME))
+$(info Android NDK home is           : $(ANDROID_NDK_HOME))
+$(info Android SDK download url is   : $(ANDROID_SDK_TOOLS_DL_URL))
+$(info Android NDK download url is   : $(ANDROID_NDK_DL_URL))
+$(info Android API level is          : $(ANDROID_API_LEVEL))
+$(info Android NDK version is        : $(ANDROID_NDK_VERSION))
+$(info JAVA_HOME is                  : $(JAVA_HOME))
+
+all: install_sdk install_ndk
+
+install_sdk: download_android_sdk extract_android_sdk update_android_sdk
+
+install_ndk: download_android_ndk extract_android_ndk
+
+download_android_sdk:
+	curl --location --progress-bar --continue-at - \
+	$(ANDROID_SDK_TOOLS_DL_URL) --output $(ANDROID_SDK_TOOLS_ARCHIVE)
+
+download_android_ndk:
+	curl --location --progress-bar --continue-at - \
+	$(ANDROID_NDK_DL_URL) --output $(ANDROID_NDK_ARCHIVE)
+
+# Extract android SDK and remove the compressed file
+extract_android_sdk:
+	mkdir -p $(ANDROID_SDK_HOME) \
+	&& unzip -q $(ANDROID_SDK_TOOLS_ARCHIVE) -d $(ANDROID_SDK_HOME) \
+	&& rm -f $(ANDROID_SDK_TOOLS_ARCHIVE)
```

Comment:
Minor for later.
I'm wondering if we do want to clean it or use it as a target for not re-downloading it next time. So basically the idea wold be to have something like:
```mk
$(ANDROID_SDK_TOOLS_ARCHIVE):
	curl --location --progress-bar --continue-at - \
	$(ANDROID_SDK_TOOLS_DL_URL) --output $(ANDROID_SDK_TOOLS_ARCHIVE

download_android_sdk: $(ANDROID_SDK_TOOLS_ARCHIVE)
```
That way running the target twice won't run the download twice if already present. Same story for other targets/dependencies

Reply:
I was thinking in keeping docker's layer smallest as possible...since we don't to re use the removed files right now...so ...maybe better to change/test this if we need it, right? :wink:

Reply:
Yes let's keep it that way :smile: